source/javascripts/hooks/api/useMonolithApiCallback.ts

```diff
@@ -5,12 +5,20 @@ export interface MonolithError {
 	error_msg: string;
 }
 
-export default function useMonolithApiCallback<T>(url: string, init?: RequestInit): FetchResponse<T, MonolithError> {
-	return useFetchCallback<T, MonolithError>(url, {
-		...init,
-		headers: {
-			"X-CSRF-TOKEN": getCookie("CSRF-TOKEN"),
-			...init?.headers
-		}
-	});
+export default function useMonolithApiCallback<T>(
+	url: string,
+	init?: RequestInit,
+	parser?: any
+): FetchResponse<T, MonolithError> {
+	return useFetchCallback<T, MonolithError>(
+		url,
+		{
+			...init,
+			headers: {
+				"X-CSRF-TOKEN": getCookie("CSRF-TOKEN"),
+				...init?.headers
+			}
+		},
+		parser as any
```

Comment:
I'm a little concerned about this. `useFetchCallback` passes the body to this function, so without any type safety, I can essentially pass something to a string or something. Do you think this is worth spending some extra effort on?

Reply:
Sure, I'll fix the typing